daemon/containerd/image.go

```diff
@@ -67,6 +68,12 @@ func (i *ImageService) GetImage(ctx context.Context, refOrID string, options ima
 			}
 			return errdefs.System(fmt.Errorf("failed to read config of the manifest %v: %w", img.Target().Digest, err))
 		}
+
+		if options.Platform != nil && !platform.Match(ociimage.Platform) {
+			log.G(ctx).WithField("platform", cplatforms.Format(ociimage.Platform)).Info("no match")
+			return nil
+		}
+
```

Comment:
So I added this, but not sure if that's correct (or if the current behavior was _intentional_ for some reason 🤔)

Reply:
Yep, its correct.

Lack of `Match` was an oversight that didn't really had any visible effect because without overriding the platform, the default was to pick the default host platform, or "whatever comes first" if its no present (`AllPlatformsWithPreference`).

Reply:
The sorting is still needed though, specifically for matchers (like `AllPlatformsWithPreference`) which have some preferences about the platform that should be chosen.

Reply:
Yup; sorting looked ok indeed.

Thanks for looking! I'll split these changes to a separate branch and PR 👍